### ISSUE TITLE
Context menus in Firefox read inconsistently with selective UIA handling enabled

### ISSUE THREAD

**Issue by MarcoZehe** [nvaccess/nvda#11354]
### Steps to reproduce:

1. Install an Alpha version of NVDA.
2. Enable the Advanced/UIA/Selective UIA Event Registration option. See #11214 .
3. In Firefox, enable the multiple account containers option in Preferences.
4. open the context menu for any link.
5. Select the menu item named Open Link in New Container", which has a sub menu.
6. Press RightArrow or Enter to open that sub menu. Now, one of two things can happen.

### Actual behavior:

* NVDA only speaks the newly focused item. Normally, it speaks that the sub menu is active first, then the item. Or:
* NVDA only speaks the name of the sub menu, but not the newly focused item.

### Expected behavior:

NVDA should behave consistently with regards to these sub menus. This isn't even an UIA application.

#### NVDA installed/portable/running from source:

Yes.

#### NVDA version:

version alpha-20514,e87d8ab1

#### Windows version:

10.0 version 2004.

#### Name and version of other software in use when reproducing the issue:

Firefox 78, 79 beta, or 80 nightly.

#### Other information about your system:

Braille also updates only if the first of the two possible outcomes happens. In the other case, it doesn't show the active menu item either.

#### Does the issue still occur after restarting your computer?

Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No, because this is specifically testing that new feature of selective UIA Event Registration.

#### If addons are disabled, is your problem still occuring?

Yes.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?

No, not needed because with Selective UIA Event Registration disabled, the result is consistent.

**Comment by leonardder** [comment]
May be a silly question, but... I can't seem to find this option in Firefox 78. May be I"m missing a prerequisite to see this option? Search for "container" in the options doesn't yield any result.

**Comment by MarcoZehe** [comment]
Sorry, you appear to still need to install an add-on for this option to show up. I've been using this for so long that I forgot. [Here](https://support.mozilla.org/en-US/kb/containers) is the info.

**Comment by MarcoZehe** [comment]
Oh, I just found another breakage that is very likely related. When initiating any download in Firefox, the whole download dialog is not read, only the current group item plus the radio button that is selected, is. So the Download dialog title and the fact that this is a dialog is skipped.

**Comment by leonardder** [comment]
What value do you have selected for Attempt to cancel speech for expired focus events?

**Comment by MarcoZehe** [comment]
It's set to "Yes". And together with that other setting, selective focus events, it seems to trigger this bug.

**Comment by leonardder** [comment]
> It's set to "Yes". And together with that other setting, selective focus events, it seems to trigger this bug.

So either when you disable one or the other, the bug is gone? That's rather weird. As you noted correctly, there is or should be no UIA involved in FireFox at all.

**Comment by leonardder** [comment]
> ### Actual behavior:
> 
>     * NVDA only speaks the newly focused item. Normally, it speaks that the sub menu is active first, then the item. Or:

Hmm. Even when I disable both selective UIA event registration and cancellable speech, I can reproduce this on my system. I can, for example, reproduce it 80% of the time in the languages submenu of the text area context menu.
I have the feeling that this is more like a timing issue, and that disabling selective UIA event registration makes this more prevalent on your system as the NVDA core is less busy.

**Comment by MarcoZehe** [comment]
Correct. But I only tried the latter, the cancellable speech setting, after you pointed me to it. I had set it to Yes a while ago to run with it to see if I spotted something weird. Yesterday, I reproduced this only by enabling or disabling selective UIA event registration. My machine is a 2017 Dell XPS15-9560, a 7th Generation Intel Core I7 quad-core processor with an SSD and plenty of RAM. So a quite fast machine, but not the newest model. And Windows 10 on the 2004 feature update, no insider builds.

**Comment by seanbudd** [comment]
Is this still a reproducible issue with the latest NVDA and Firefox?

**Comment by Adriani90** [comment]
No updates provided upon request since more than 3 years, closing as abandoned. If you are still having this issue, please comment with updates and we can reopen.
